Added nozzle options

fern-aviles opened the pull request:
What's new
-
- Users can now see a list of all nozzles for their selected product
- Users can see which nozzles can be selected depending on their specifications

Set Up
-
- Clone repo
- In your command line:
  - run npm install to install necessary packages
  - run npm run dev to run the project in your web browser

On your browser, go to localhost:5173 and you should see a page with a black line on the top left and text reading "5 feet" right under it.

Testing
-
- Adding a product on the page should add a nozzle options list on the left
- The nozzles list should be switching between red and black text
  - The black colored nozzles work with the current settings of the product
  - The red colored nozzles do not work with the current settings of the product
- Selecting a nozzle should turn green and whenever the Water radius is no longer within that nozzle's range, it is deselected
- On the console, there should be details about the chosen nozzle whenever the user finishes moving the radius or whenever they select a nozzle

Commit:
Added nozzle options feature

- Users can now see all available nozzles for their selected product
- Users can see which nozzles can be selected depending on their
  specifications

Commit:
Updated code documentation

scott-allison commented:
@fern-aviles, the PGP Ultra needs to go a full 360º. Please revise. Nozzle data checks out, nice job.

scott-allison reviewed (changes requested):
PGP Ultra max arc should be 360

Commit:
Increased product 862 max radius

Commit:
All controllers change available nozzles

- Before only the mid controller changed the available nozzles
- Start and end controllers update the nozzles list depending on whether
  the angle still fits for each nozzle

fern-aviles commented on src/util/data.json:
Increased product 862 max **arc**

scott-allison replied:
Looking at the behavior of 360º, it's not functioning very well when you want to set it exactly to 360. Can we make anything over 355º snap to 360º, and on the opposite side, anything under 5º is 360º. In the case of PGP Ultra with a minimum arc of 50º, only when the users mouse is dragging within 355º and 5º, it should stay at 360º. I believe this "padding" for 360º will help users achieve this full-circle mode, since this is pretty common. @fern-aviles

Commit:
Refactored if statements

Commit:
Refactored if statements

scott-allison commented on src/util/data.json:
```diff
@@ -1,1002 +1,1024 @@
+                }
+            },
+            "mpr-25": {
+                "90°": {
```
90º is the correct angle, but it should be labeled Q. Here's the nozzle rack. Q for quarter (90º), T for third (120º), H for half (180º) and F for full (360º). This applies to all MPR nozzle options.
![Screenshot 2024-07-26 at 9 23 46 AM](https://github.com/user-attachments/assets/4ea29dbd-a012-4535-ab32-87b6162d5f83)

Commit:
Changed degrees to QTHF values

Commit:
Added snap to full circle feature
